include/wlr/types/wlr_xdg_foreign_v1.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_XDG_FOREIGN_V1_H
+#define WLR_TYPES_WLR_XDG_FOREIGN_V1_H
+
+#include <wayland-server-core.h>
+#include <wlr/types/wlr_xdg_foreign_registry.h>
+
+struct wlr_xdg_foreign_v1 {
+	struct wl_global *exporter_global;
+	struct wl_global *importer_global;
+
+	struct {
+		struct wl_global *global;
+		struct wl_list objects; // wlr_xdg_exporter_v1::link or wlr_xdg_importer_v1::link
+	} exporter, importer;
+
+	struct wl_listener foreign_registry_destroy;
+	struct wl_listener display_destroy;
+
+	struct wlr_xdg_foreign_registry *registry;
+
+	struct {
+		struct wl_signal destroy;
+	} events;
+
+	void *data;
+};
+
+struct wlr_xdg_exporter_v1 {
```

Comment:
I still don't think we need this intermediate struct. We can just store the imported/exported objects in the `wlr_xdg_foreign_v1`.

Reply:
You are right! I misunderstood your previous comment. This change removed a lot of unnecessary code.